paimon-core/src/main/java/org/apache/paimon/operation/KeyValueFileStoreWrite.java

```diff
@@ -356,13 +361,26 @@ private <T> LookupLevels<T> createLookupLevels(
                         cacheManager,
                         new RowCompactedSerializer(keyType).createSliceComparator());
         Options options = this.options.toConfiguration();
+        InternalRow.FieldGetter[] getters = partitionType.fieldGetters();
+        StringBuilder builder = new StringBuilder();
+        for (InternalRow.FieldGetter getter : getters) {
+            Object part = getter.getFieldOrNull(partition);
+            if (part != null) {
+                builder.append(part);
+            } else {
+                builder.append("null");
+            }
+            builder.append("-");
+        }
+        String prefix = String.format("%s-%s", builder, bucket);
```

Comment:
I am concerned that it may be too long. Can we use its hash directly as a prefix?

Reply:
We can know files are come from the same partition.

Reply:
I think the length is acceptable, here is an actual case. Hash prefix is not intuitive.

<img width="900" alt="image" src="https://github.com/user-attachments/assets/5ef03b54-daad-47ca-89c8-7b6cb3a3b9b9">

Reply:
Can we cut it off? For example, if it exceeds 20 characters, do not add any more

Reply:
Sure, it's safer